validator/sawtooth_validator/execution/executor.py

```diff
@@ -301,3 +341,21 @@ def __setitem__(self, key, value):
     def __delitem__(self, key):
         with self._lock:
             del self._waiters[key]
+
+    def __len__(self):
+        with self._lock:
+            return len(self._waiters)
+
+    def values(self):
+        """
+        Returns (list of _Waiter): All the _Waiter objects.
+        """
+        with self._lock:
+            return self._waiters.values()
```

Comment:
In python 3, this will return a dict_values object, so this code is probably not thread-safe currently.  You probably want: return list(self._waiters.values()).